src/box/lua/connpool.lua

```diff
@@ -149,6 +149,26 @@ local function is_roles_match(expected_roles, present_roles)
     return true
 end
 
+local function is_sharding_roles_match(expected_roles, present_roles)
+    if expected_roles == nil or next(expected_roles) == nil then
+        return true
+    end
+    if present_roles == nil or next(present_roles) == nil then
+        return false
+    end
+
+    local roles = {}
+    for _, present_role_name in pairs(present_roles) do
+        roles[present_role_name] = true
+    end
+    for _, expected_role_name in pairs(expected_roles) do
+        if roles[expected_role_name] == nil then
+            return false
+        end
+    end
+    return true
+end
```

Comment:
It is a copy of `is_roles_match`. I would just call `is_roles_match`.

Reply:
Thank you. Applied.